aya/src/programs/lirc_mode2.rs

```diff
@@ -60,9 +61,9 @@ impl LircMode2 {
     /// Attaches the program to the given lirc device.
     ///
     /// The returned value can be used to detach, see [LircMode2::detach].
-    pub fn attach<T: AsRawFd>(&mut self, lircdev: T) -> Result<LircLinkId, ProgramError> {
+    pub fn attach<T: AsFd>(&mut self, lircdev: T) -> Result<LircLinkId, ProgramError> {
```

Comment:
I wonder if this should actually just take an `OwnedFd`, which would remove our need to clone it and it'd be more obvious that we are taking ownership of the file descriptor whereas now it is not clear IMO who owns the file descriptor when calling for `attach` (if we do it here we may also want/need to do in other attach functions to be consistent though).